plugins/exporter.koplugin/target/joplin.lua

```diff
@@ -0,0 +1,343 @@
+local BD = require("ui/bidi")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local http = require("socket.http")
+local json = require("json")
+local logger = require("logger")
+local ltn12 = require("ltn12")
+local socketutil = require("socketutil")
+local T = require("ffi/util").template
+local _ = require("gettext")
+
+-- joplin exporter
+local JoplinExporter = require("base"):new {
+    name = "joplin",
+    is_remote = true,
+    notebook_name = _("KOReader Notes"),
+}
+
+local function makeRequest(url, method, request_body)
+    local sink = {}
+    local request_body_json = json.encode(request_body)
+    local source = ltn12.source.string(request_body_json)
+    socketutil:set_timeout(socketutil.LARGE_BLOCK_TIMEOUT, socketutil.LARGE_TOTAL_TIMEOUT)
+    http.request{
+        url     = url,
+        method  = method,
+        sink    = ltn12.sink.table(sink),
+        source  = source,
+        headers = {
+            ["Content-Length"] = #request_body_json,
+            ["Content-Type"] = "application/json"
+        },
+    }
+    socketutil:reset_timeout()
+
+    if not sink[1] then
+        return nil, "No response from Joplin Server"
+    end
+
+    local response = json.decode(sink[1])
+
+    if response.error then
+        return nil, response.error
+    end
+
+    return response
+end
+
+local function ping(ip, port)
+    local sink = {}
+    http.request{
+        url =  "http://"..ip..":"..port.."/ping",
```

Comment:
I just noticed this at random, but that `http` doesn't feel like it'd necessarily always work these days?

Reply:
We should add a protocol field in the settings, default to `http`.

Reply:
I'm afraid it doesn't work that way.

The web clipper is mostly unsecure and intended to be used in localhost. There's no way to tweak the joplin to listen to other hosts. That's why the workarounds are mentioned in "help".

The main intent of the web clipper is to sync stuff from chrome/firefox extensions. Other apps running on the same host can interact with it too. 

That's what we do. Use the "unsecure" localhost API to write highlights.